src/main/java/com/hivemq/mqtt/services/PublishPollServiceImpl.java

```diff
@@ -122,29 +117,25 @@ public void pollMessages(final @NotNull String client, final @NotNull Channel ch
                             topic.isRetainAsPublished(), topic.getSubscriptionIdentifier(), channel);
                 }
             } catch (final ExecutionException e) {
-                log.error("Exception while reading shared subscriptions for client " + client, e);
+                log.error("Exception while reading shared subscriptions for client {}", client, e);
             }
 
         } else {
             pollInflightMessages(client, channel);
         }
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public void pollNewMessages(final @NotNull String client) {
-        final Channel channel = channelPersistence.get(client);
-        if (channel == null) {
-            return; // client is disconnected
+        final ClientConnection clientConnection = connectionPersistence.get(client);
+        if (clientConnection == null
+                || clientConnection.getClientState() == ClientState.DISCONNECTING
+                || clientConnection.getClientState().disconnected()) {
```

Comment:
just to make sure - this is a change from some other PR, not from this refactoring?